ttnn/cpp/ttnn/operations/full/device/full_device_operation.cpp

```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "full_device_operation.hpp"
+
+#include "ttnn/tensor/tensor.hpp"
+
+namespace ttnn::operations::full {
+void FullOperation::validate_inputs(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    auto any = tensor_args.any;
+    TT_FATAL(any.storage_type() == StorageType::DEVICE, "Full operation error: Any tensor must be on device");
+    TT_FATAL(
+        operation_attributes.memory_config.memory_layout == TensorMemoryLayout::INTERLEAVED,
+        "Full operation error: Not currently supporting sharding");
+    TT_FATAL(
+        operation_attributes.layout == Layout::TILE, "Full operation error: Not currently supporting row major layout");
+
+    const auto shape = operation_attributes.shape;
+
+    TT_FATAL(
+        shape.size() > 1,
+        "Full operation error: Shape size must be greater than 1, but got shape size = {}",
+        shape.size());
+
+    for (int i = 0; i < shape.size(); i++) {
+        TT_FATAL(
+            shape[i] > 0,
+            "Full operation error: Invalid shape at index {}. Each dimension of the shape must be greater than 0, but"
+            "got shape[{}] = {}",
+            i,
+            i,
+            shape[i]);
+    }
+}
+
+FullOperation::program_factory_t FullOperation::select_program_factory(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    return ProgramFactory{};
+}
+
+void FullOperation::validate_on_program_cache_miss(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    validate_inputs(operation_attributes, tensor_args);
+};
+
+void FullOperation::validate_on_program_cache_hit(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    validate_inputs(operation_attributes, tensor_args);
+};
+
+FullOperation::shape_return_value_t FullOperation::compute_output_shapes(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    auto shape = operation_attributes.shape;
+    return SimpleShape(std::move(const_cast<std::vector<uint32_t>&>(shape)));
```

Comment:
maybe just
```suggestion
    return SimpleShape(operation_attributes.shape);
```

Reply:
I wasn't able to directly create a SimpleShape that way before the constructor was updated. I've modified my code.